src/server/qtquick/private/wquickforeigntoplevelmanagerv1.cpp

```diff
@@ -64,6 +70,41 @@ class WQuickForeignToplevelManagerV1Private : public WObjectPrivate {
             handle->outputLeave(output->handle());
         }));
 
+        connection.push_back(QObject::connect(handle.get(),
+                             &QWForeignToplevelHandleV1::requestActivate,
+                             surface,
+                             [surface](wlr_foreign_toplevel_handle_v1_activated_event *event) {
+                                 surface->setActivate(event->toplevel->state & WLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_ACTIVATED);
```

Comment:
This job should do in the Compositor, not in Waylib.
You can emit a signal here.

Same for the other request events.